tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/QueryFixerMain.java

```diff
@@ -1,227 +1,75 @@
-    if (credentialPath == null) {
-      return BigQueryOptions.newBuilder().setProjectId(projectId).build();
-    } else {
-      // TODO: should support this in near future.
-      System.out.println("Customized credential path is not supported");
-      System.exit(1);
-      return null;
-    }
-  }
-
-  private static FixResult fixQueryInFullInteractMode(
-      AutomaticQueryFixer queryFixer, String query) {
-    return queryFixer.fix(query);
-  }
-
-  private static void printFixResult(String query, FixResult fixResult, String outputFormat) {
-    if (outputFormat == null || outputFormat.equalsIgnoreCase(NATURAL_OUTPUT)) {
-      System.out.println("Input query: " + query);
-      printFixResultInCommandLine(fixResult);
-    } else if (outputFormat.equalsIgnoreCase(JSON_OUTPUT)) {
-      printFixResultAsJson(fixResult);
-    } else {
-      System.out.println("Output format (-o) is incorrect. Use --help for usage.");
-      System.exit(1);
+  /** Create the BigQueryOption based on user-input credentials path and project ID. */
+  private static BigQueryOptions buildBigQueryOptions(String credentialsPath, String projectId) {
+    if (projectId == null) {
+      System.out.println("Project ID should not be null. Please provide it through the flag -p.");
+      printHelpAndExit();
     }
-  }
-
-  private static void printFixResultAsJson(FixResult fixResult) {
-    Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
-    System.out.println(gson.toJson(fixResult));
-  }
-
-  private static void printFixResultInCommandLine(FixResult fixResult) {
-    if (fixResult.getStatus() == FixResult.Status.NO_ERROR) {
-      System.out.println("The input query is valid. No errors to fix.");
-      return;
-    }
-
-    System.out.println("The query has an error: " + fixResult.getError());
 
-    if (fixResult.getStatus() == FixResult.Status.FAILURE) {
-      System.out.println("Failed to fix the input query.");
-      return;
-    }
-
-    System.out.println("It can be fixed by the approach: " + fixResult.getApproach());
-    printFixOptions(fixResult.getOptions());
-  }
-
-  private static void printFixOptions(List<FixOption> options) {
-    System.out.println();
-    int count = 1;
-    for (FixOption option : options) {
-      System.out.println(String.format("%d. Option: %s", count++, option.getDescription()));
-      System.out.println(String.format("   Fixed query: %s", option.getFixedQuery()));
-      System.out.println();
+    // If no credentials is provided, the program uses:
+    // <ol>
+    // <li> the default path in the Env variable: GOOGLE_APPLICATION_CREDENTIALS.
+    // <li> Default credentials location (OS dependent). For example,
+    // "~/.config/gcloud/application_default_credentials.json" in Linux. Usually, it should be same
+    // path of the credentials you create through calling the command "gcloud auth
+    // application-default login".
+    // </ol>
+    if (credentialsPath == null) {
```

Comment:
Will it be empty too?

Reply:
Yes, similar as the `projectID`. If it is an empty string, it should be considered as an incorrect path to a credentials. It will be handled inside BigQueryOptions.